consensus/istanbul/validator/default.go

```diff
@@ -251,16 +291,27 @@ func (valSet *defaultSet) AsData() *istanbul.ValidatorSetData {
 	}
 }
 
+func (valSet *defaultSet) AsDataWithCache() *istanbul.ValidatorSetDataWithCache {
```

Comment:
do we need this method? it seems the whole reason for the change is for snapshot to cache and persist validators with the uncompressed keys. Now, snapshot's marshallJSON doesn't use this method, nor it marshalls directly the set, instead it marshalls the validatorList only.